azurerm/internal/services/eventgrid/resource_arm_eventgrid_event_subscription.go

```diff
@@ -18,6 +20,46 @@ import (
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
+func getEnpointTypes() []string {
+	return []string{"webhook_endpoint", "storage_queue_endpoint", "eventhub_endpoint", "hybrid_connection_endpoint", "service_bus_queue_endpoint", "service_bus_topic_endpoint", "azure_function_endpoint"}
+}
+
+// RemoveFromStringArray removes all matching values from a string array
+func RemoveFromStringArray(elements []string, remove string) []string {
```

Comment:
Could we move this to the utils package?

Reply:
> Hey @jrauschenbusch, thanks for this PR! However typically we try not to combine so many changes into one PR. Could you split this out into an SDK upgrade PR and then the rest of the changes into a couple different PRs? Large PRs like this take far longer to review and merge thus are more likely to sit in the qeue for a while. Thanks!

@katbyte Thanks for your response. I already had this feeling that the PR is ways too large. I try to split up the PR into multiple smaller ones.

My proposal:
- PR for Basic Event Grid API Update to the latest Preview version (https://github.com/Azure/azure-sdk-for-go/tree/master/services/preview/eventgrid/mgmt/2020-04-01-preview/eventgrid)
- PR for SDK util functions
- PR for new resource `azurerm_eventgrid_topic_domain`
- PR for extended resource `azurerm_eventgrid_topic`
- PR for extended resource `azurerm_eventgrid_event_subscription`
- Dedicated PR for `advanced_filters` in resource `azurerm_eventgrid_event_subscription`

Reply:
You can attach the util functions into the first PR requiring them but otherwise sounds good!